lib/init.coffee

```diff
@@ -1,61 +1,59 @@
-{BufferedProcess, CompositeDisposable} = require 'atom'
+{ BufferedProcess, CompositeDisposable } = require 'atom'
+helpers = require 'atom-linter'
 
 module.exports =
-  config:
-    executablePath:
-      type: 'string'
-      title: 'The path where Roodi is located'
-      default: 'roodi'
   activate: ->
-    console.log 'activate linter-roodi'
+    console.log 'activate linter-roodi' if atom.inDevMode()
     @subscriptions = new CompositeDisposable
     @subscriptions.add atom.config.observe 'linter-roodi.executablePath',
-      (executablePath) =>
+      (executablePath) ->
         @executablePath = executablePath
   deactivate: ->
     @subscriptions.dispose()
   provideLinter: ->
     provider =
+      name: 'roodi'
       grammarScopes: ['source.ruby', 'source.ruby.rails', 'source.ruby.rspec']
       scope: 'file' # or 'project'
-      lintOnFly: true # must be false for scope: 'project'
-      lint: (textEditor) =>
-        return new Promise (resolve, reject) =>
+      lintsOnChange: true # must be false for scope: 'project'
+      lint: (textEditor) ->
+        return new Promise (resolve, reject) ->
           filePath = textEditor.getPath()
-          output = ""
+          output = ''
           process = new BufferedProcess
             command: @executablePath
-            args: [filePath, '--json']
+            args: [filePath]
             stdout: (data) ->
               output = data
             exit: (code) ->
               return resolve [] if code is 0
 
+              console.log(output)
+
               lines = output.split("\n")
-              errorPattern = /.+?:(\d+) - (.+)\[/
+              errorPattern = /.+?:(\d+) - (.*)\x1B\[[0-?]/
```

Comment:
A comment wouldn't hurt, but it's probably not necessary since if it ever needs to be changed the person changing it would like discover the reason for them soon enough.